### PR TITLE
Remove unused/nop/questionable code

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**

### CODE REVIEW

File: src/utilcode/util.cpp [hunk 1]
```diff
@@ -2279,32 +2216,6 @@ HRESULT GetImageRuntimeVersionString(PVOID pMetaData, LPCSTR* pString)
     
     // Header data starts after signature.
     *pString = (LPCSTR) pSig->pVersion;
-    if(*pString) {
```
Comment (AaronRobinsonMSFT) [inline src/utilcode/util.cpp]: This branch seems like a very old code path that ensures Version strings in metadata are valid. It seems like a legacy code path for early development of the CLR.

@jkotas I debugged through this via crossgen and it basically just validates the embedded version string is acceptable and if not replaced it with `v1.0.3705` 😖. Is there anything more I could do to verify this change is okay?

File: src/vm/profilinghelper.cpp [hunk 2]
```diff
@@ -1049,14 +1049,6 @@ HRESULT ProfilingAPIUtility::LoadProfiler(
 
         _ASSERTE(profilerCompatibilityFlag == kEnableV2Profiler);
 
-        // To prevent V2 profilers from AV, once a V2 profiler is already loaded by a V2 rutnime in the process, 
-        // V4 runtime will not try to load the V2 profiler again.
-        if (IsV2RuntimeLoaded())
-        {
-            LogProfInfo(IDS_PROF_V2PROFILER_ALREADY_LOADED, wszClsid);
```
Comment (AaronRobinsonMSFT) [inline src/vm/profilinghelper.cpp]: I removed the display, but not the actual resource. What is the process here to remove the resource string? Just edit the resource files manually?

Comment (AaronRobinsonMSFT) [reply src/vm/profilinghelper.cpp]: cc @noahfalk

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness

File: src/utilcode/util.cpp (hunk 1 above)
Comment (jkotas) [reply src/utilcode/util.cpp]: I think deleting this is fine. CoreCLR does not use the metadata string for anything.

File: src/vm/profilinghelper.cpp (hunk 2 above)
Comment (jkotas) [reply src/vm/profilinghelper.cpp]: > Just edit the resource files manually?

Yes.

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot  test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test

**Pull request merged**